benchmarks/index.html

```diff
@@ -1,21 +1,103 @@
-        let htmlString = '<h1>Performance tests</h1><ul>';
-        for (let file of data) {
-          if (file.type === 'dir') {
-            htmlString += `<li>${file.name}</li><ul>`;
-            htmlString += `<li><a href="${file.name}/files/upload/index.html">Files upload</a></li>`;
-            htmlString += '</ul>';
-          }
+<body>
+<script src="https://cdn.jsdelivr.net/npm/chart.js@2.8.0"></script>
+<div class="chart-container" style="position: relative; height:30vh; width:80vw">
+    <canvas id="docker_single_upload"></canvas>
+</div>
+<script>
+    /**
+     * Compare strings with numeric versions
+     * @param one The one string
+     * @param another The another string
+     * @returns {number} The result of comparison
+     */
+    function compareDottedString(one, another) {
+        let a1 = one.split('.');
+        let b1 = another.split('.');
+        let len = Math.max(a1.length, b1.length);
+        for (var i = 0; i < len; i++) {
+            let _a = +a1[i] || 0;
+            let _b = +b1[i] || 0;
+            if (_a === _b) continue;
+            else return _a > _b ? 1 : -1
         }
-        htmlString += '</ul>';
-        document.getElementsByTagName('body')[0].innerHTML = htmlString;
-      })()
-    </script>
-  </body>
+        return 0;
+    }
+
+    (async () => {
+        let repository = 'artipie/artipie';
+        let branch = 'gh-pages';
+        const response = await fetch(`https://api.github.com/repos/${repository}/contents/benchmarks?ref=${branch}`);
+        const data = await response.json()
+
+        // array of benchmarks results and versions
+        let results = (await Promise.all(
+                data.filter(el => el.type === 'dir')
+                    .map(el => {
+                        let version = el.name;
+                        return fetch(
+                            `https://raw.githubusercontent.com/${repository}/${branch}/benchmarks/${version}/benchmark-results.json`
+                        ).then(response => {
+                            if (!response.ok) {
+                                throw new Error(`Can't find results for version ${version}`)
+                            }
+                            return response.json()
+                        }).then(json => {
+                            return {
+                                "version": version,
+                                "results": json
+                            }
+                        }).catch(er => {
+                        })
+                    })
+            )
+        ).filter(el => el !== undefined)
+            .sort((one, another) => compareDottedString(one.version, another.version))
+
+
+        // Draw Chart
+        let labels = results.map(json => json.version)
+        let dt = results.map(json => json.results.docker["single-upload"].artipie.images[0].ubuntu)
+        let ctx = document.getElementById('docker_single_upload').getContext('2d');
```

Comment:
@Sammers21 I don't think we need to hard-code Docker benchmarks in code, how we will switch between benchmarks when we have many of them?

Reply:
@g4s8, it is the first step. Anything else will come later